README.md

```diff
@@ -80,17 +80,47 @@ $ cd parsec
 $ RUST_LOG=info cargo run
 ```
 
-Parsec Client Libraries can now communicate with the service. For example using the Rust Test client,
+Parsec Client Libraries can now communicate with the service. For example using the Rust client,
 RSA signatures can be done as follows:
 ```rust
-use parsec_client_test::TestClient;
-
-let mut client = TestClient::new();
+use parsec_client::BasicClient;
+use parsec_client::auth::AuthenticationData;
+use parsec_client::core::ProviderID;
+use parsec_client::core::psa_algorithm::{Algorithm, AsymmetricSignature, Hash};
+use parsec_client::core::psa_key_attributes::{KeyAttributes, KeyType, KeyPolicy, UsageFlags};
+use sha2::Sha256;
+
+let app_identity_ = AuthenticationData::AppIdentity(String::from("my-app"));
+let client = BasicClient::new(app_identity, ProviderID::Tpm);
+let sign_alg = AsymmetricSignature::RsaPkcs1v15Sign {
+    hash_alg: Hash::Sha256,
+};
+let key_attrs = KeyAttributes {
+    key_type: KeyType::RsaKeyPair,
+    key_bits: 1024,
+    key_policy: KeyPolicy {
+        key_usage_flags: UsageFlags {
+            sign_hash: true,
+            verify_hash: true,
+            sign_message: true,
+            verify_message: true,
+            export: true,
+            encrypt: false,
+            decrypt: false,
+            cache: false,
+            copy: false,
+            derive: false,
+        },
+        key_algorithm: Algorithm::AsymmetricSignature(sign_alg),
+    },
+};
 let key_name = String::from("🔑 What shall I sign? 🔑");
-client.generate_rsa_sign_key(key_name.clone()).unwrap();
-let signature = client.sign(key_name,
-                            String::from("Platform AbstRaction for SECurity").into_bytes())
-                      .unwrap();
+client.psa_generate_key(key_name.clone(), key_attrs).unwrap();
```

Comment:
These key name cloning are so annoying :/ I am wondering if we should make the functions take a reference and clone when creating the requests. If we do I am not sure if that should be part of this layer or above.

Reply:
I know... we can either do that or make a Rust type for key names that we implement `Copy` for